src/Orleans.Core/Configuration/OptionLogger/IOptionsLogger.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Orleans
+{ 
+    internal class ClientOptionsLogger : OptionsLogger, ILifecycleParticipant<IClusterClientLifecycle>
+    {
+        private int ClientOptionLoggerLifeCycleRing = 1000;
+        public ClientOptionsLogger(ILogger<ClientOptionsLogger> logger, IServiceProvider services)
+            : base(logger, services)
+        {
+        }
+
+        public void Participate(IClusterClientLifecycle lifecycle)
+        {
+            lifecycle.Subscribe(ClientOptionLoggerLifeCycleRing, this.OnRuntimeInitializeStart);
+        }
+
+        public Task OnRuntimeInitializeStart(CancellationToken token)
+        {
+            this.LogOptions();
+            return Task.CompletedTask;
+        }
+    }
+
+    public abstract class OptionsLogger 
+    {
+        private ILogger logger;
+        private IServiceProvider services;
+        protected OptionsLogger(ILogger logger, IServiceProvider services)
+        {
+            this.logger = logger;
+            this.services = services;
+        }
+        public void LogOptions()
+        {
+            this.logger.LogInformation("Starting with following configuration:");
```

Comment:
it provides information about : as follows is the configuration info. Otherwise it will be confusing to user to figure out what are those following this line . 

What else do you think it need ?

Reply:
That's the thing about log data, it's not necessarily well ordered by the log.  This line could be followed by thousands of other lines before any option information is written.  The only guarantee (roughly) is that somewhere in the future information related to this line will start showing up.  That's of little value.  I think this line should be removed, or minimally be verbose.